client/app/bundles/course/assessment/submission/actions/submissions.js

```diff
@@ -172,32 +169,29 @@ export function downloadStatistics(type) {
     dispatch({ type: actionTypes.DOWNLOAD_STATISTICS_REQUEST });
 
     const handleSuccess = (successData) => {
-      window.location.href = successData.redirect_url;
+      window.location.href = successData.redirectUrl;
       dispatch({ type: actionTypes.DOWNLOAD_STATISTICS_SUCCESS });
       dispatch(setNotification(translations.downloadRequestSuccess));
     };
 
-    const handleFailure = (data) => {
+    const handleFailure = (error) => {
       const message =
-        (data &&
-          data.response &&
-          data.response.data &&
-          data.response.data.error) ||
+        error?.response?.data?.error ||
```

Comment:
Since `error?.response?.data?.error` evaluates to `undefined`, use `??` instead of `||`. Same for `error?.message` below.

Reply:
It could be the case that `error?.response?.data?.error` or `error?.message` returns an empty string `""`. In such a case, if we use `??`, it will return the empty string right away. Even if they evaluate to `undefined` or `null`, either `??` or `||` would the value on the right.

Reply:
I see, didn't know that it could be that `error` be an empty string. Thanks for clarifying!

Reply:
Well we hope not! but just in case~